### ISSUE TITLE
Attach Error not Passed to .catch

### ISSUE THREAD

**Issue by BebeSparkelSparkel** [visionmedia/superagent#1305]
No way to catch bad file path link for .attach

```javascript
const request = require('superagent')

try {
  request
    .post('/somewhere')
    .attach('image1', './bad/path')
    .end(function(err, res){
      if (err) {
        console.error('end error')
      }
      else {
        console.log('no error')
      }
    })
} catch(err) {
  console.error('try/catch error')
}
```

Yields the following:
```shell
events.js:182
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open './bad/path'
```

So it's impossible for me to catch this and when using supertest leaves my test hanging with an unhandled promise exception.

**Comment by pornel** [comment]
I think it's because the `form-data` library isn't passing this error to superagent. I've filed an issue with them.

**Comment by niftylettuce** [comment]
Fixed in superagent v7.0.0.

See <https://github.com/visionmedia/superagent/releases/tag/v7.0.0> for more insight.

We've also published this deprecation notice:

```sh
Please upgrade to v7.0.0+ of superagent.  We have fixed numerous issues with streams, form-data, attach(), filesystem errors not bubbling up (ENOENT on attach()), and all tests are now passing.  See the releases tab for more information at <https://github.com/visionmedia/superagent/releases>. Thanks to @shadowgate15 (aka Super Agent Taylor), @spence-s, and @niftylettuce.
```
